### PR TITLE
include --glob flag in rimraf

### PR DESCRIPTION
**Pull request by connorskees**
Running `lerna run clean --scope=@fluidframework/merge-tree` currently crashes with the error `Error: Illegal characters in path.`. Adding the `--glob` flag to `rimraf` fixes this crash. I've only verified this change works for the merge-tree package, but the change should be straightforward for the other packages.

### CODE REVIEW

**Comment by Abe27342** [comment]
looks good. only 2 concerns here:

1. Do we need a policy-check update? would rather not add 153 warnings :)
2. Does this make the experience work on unix/linux?

probably just need someone with a mac or linux dev machine (maybe also WSL for good measure) to confirm 2nd one.


...and while I was typing my comment:

3. what's going on with our link checker? (I'm assuming that's completely unrelated to this PR, but i'll investigate)

**Comment by Abe27342** [comment]
answer to (3) is that https://fluidframework.com has a link to the fluidframework twitter (which... I guess is a thing) in the footer, and the linkchecker appears to give an error for each page on fluidframework.com. so the output makes sense assuming twitter was down for a moment.

**Comment by Abe27342** [comment]
> I'll double check on my personal linux machine that the command removes the right files.
> 
> For the first point, what is a policy-check update?

We have a lot of 'handlers' implemented that ensure our package.json scripts stay relatively uniform (i.e. alphabetical, reasonable deps installed, have a readme, etc. that kind of thing). I wasn't sure if we had one that enforced the contents of the `clean` script were reasonable, but I checked just now and I don't see any reference that looks troublesome. There's e.g. some logic which invokes the clean script in `fluidBuild.ts`, and some which verifies build scripts are accompanied by clean scripts in `fluidPackageCheck.ts`, but that's about it.

(I should mention that both of those files are in fluid-build and not actually policy-check; if you wanted to look over the handlers we have implemented open the file `fluidCase.ts` and look around that folder)

**Review by Abe27342 — APPROVED** [review]
LGTM once you confirm this doesn't break the experience on unix

**Comment by connorskees** [comment]
Tested on linux box, no crashes and deletes the files I would expect it to. Will merge once CI is green

**Review by tylerbutler — COMMENTED** [review]
@curtisman FYI. Will this break fluid-build's script parsing?

**Comment by tylerbutler** [comment]
We've updated to the latest build-tools in client, so as long as pnpm fluid-build works as expected with this change, ship it. :shipit:

**Comment by curtisman** [comment]
> @curtisman FYI. Will this break fluid-build's script parsing?

Nope

File: azure/package.json
Comment (CraigMacomber) [inline azure/package.json:1]: why is this file added. Seems unrelated. Bad merge?

Comment (connorskees) [reply azure/package.json:1]: Yes, bad merge. Intending to delete this before merging into main.

File: azure/packages/azure-client/package.json
Comment (CraigMacomber) [inline azure/packages/azure-client/package.json:25]: I believe before this change, this command worked on some OSes (like linux) because the shell was expanding the "*". It seems like if we are going to use --glob, for consistency it would be better to have rimraf expand it on all platforms, so we should quote the patterns to stop the shell from doing it.

**Pull request merged**
